RELEASE.md

```diff
@@ -0,0 +1,628 @@
+
+See: `Cargo.toml` (release-turborepo profile) and `.github/workflows/turborepo-release.yml`
+
+**Build Steps**:
+
+1. Install system dependencies (clang, musl-tools for Linux cross-compilation)
+2. Install Protoc (v26.x) and Cap'n Proto (code generation tools)
+3. Run: `cargo build --profile release-turborepo -p turbo --target <target>`
+4. Upload binary artifact from `target/<target>/release-turborepo/turbo*`
+
+#### Stage 5: NPM Publish
+
+**Job**: `npm-publish` (depends on all previous stages)
+
+This is the most complex stage with multiple sub-steps:
+
+##### 5a. Prepare Rust Artifacts
+
+1. Download all platform-specific binary artifacts from Stage 4
+2. Move binaries to platform-specific directories:
+   ```
+   rust-artifacts/turbo-aarch64-apple-darwin    → cli/dist-darwin-arm64/turbo
+   rust-artifacts/turbo-x86_64-apple-darwin     → cli/dist-darwin-x64/turbo
+   rust-artifacts/turbo-aarch64-unknown-linux-musl → cli/dist-linux-arm64/turbo
+   rust-artifacts/turbo-x86_64-unknown-linux-musl  → cli/dist-linux-x64/turbo
+   rust-artifacts/turbo-x86_64-pc-windows-msvc  → cli/dist-windows-x64/turbo.exe
+   ```
+
+##### 5b. Build JavaScript Packages
+
+Execute `make -C cli build` which runs `turbo build copy-schema` with filters for all JavaScript/TypeScript packages. This builds all TypeScript packages and copies the JSON schema to the appropriate locations.
+
+See: `cli/Makefile` (build target)
+
+##### 5c. Pack and Publish Native Packages
+
+Execute `turbo release-native` which invokes the `@turbo/releaser` tool.
+
+**The `@turbo/releaser` tool** (`packages/turbo-releaser/`):
+
+1. **Reads version and tag** from `version.txt`
+2. **For each platform** (6 total):
+   - Generates a native package structure with platform-specific metadata (name, os, cpu, etc.)
+   - Copies `LICENSE` and `README.md` from template
+   - For Windows: includes a `bin/turbo` Node.js wrapper script (to work around npm `.exe` stripping)
+   - Copies the prebuilt binary from `cli/dist-<os>-<arch>/`
+   - Makes the binary executable (`chmod +x` on Unix)
+   - Creates a `.tar.gz` archive
+   - Publishes to npm: `npm publish turbo-<os>-<arch>.tar.gz --tag <npm-tag>`
+
+See: `packages/turbo-releaser/` for native package generation logic
+
+**Published native packages**:
+
+- `turbo-darwin-64`
+- `turbo-darwin-arm64`
+- `turbo-linux-64`
+- `turbo-linux-arm64`
+- `turbo-windows-64`
+- `turbo-windows-arm64` (package structure only, binary not yet built)
+
+##### 5d. Pack and Publish JavaScript Packages
+
+Execute `make -C cli publish-turbo` which:
+
+1. **Packs all packages** to tarballs using `pnpm pack`
+2. **Publishes in fixed order** to npm with the appropriate dist-tag (if not `--skip-publish`)
+
+See: `cli/Makefile` (publish-turbo target)
+
+**Why fixed order?**
+
+- Prevents race conditions where dependent packages are published before their dependencies
+- Ensures `turbo` is published first so other packages can reference it
```

Comment:
I think this is wrong, the platform specific packages are published first, and then turbo (which references them)